library/Requests.php

```diff
@@ -205,20 +204,47 @@ protected static function get_transport($capabilities = array()) {
 				continue;
 			}
 
-			$result = call_user_func(array($class, 'test'), $capabilities);
+			$result = $class::test($capabilities);
 			if ($result) {
 				self::$transport[$cap_string] = $class;
-				break;
+				return $class;
 			}
 		}
-		if (self::$transport[$cap_string] === null) {
+
+		return '';
+	}
+
+	/**
+	 * Get a working transport
+	 *
+	 * @throws Requests_Exception If no valid transport is found (`notransport`)
+	 * @return Requests_Transport
+	 */
+	protected static function get_transport($capabilities = array()) {
+		$class = self::get_transport_class($capabilities);
+
+		if ($class === '') {
 			throw new Requests_Exception('No working transports found', 'notransport', self::$transports);
 		}
 
-		$class = self::$transport[$cap_string];
 		return new $class();
 	}
 
+	/**
+	 * Checks to see if we have a transport for the capabilities requested
+	 *
+	 * Supported capabilities can be found in the {@see Requests_Capability}
+	 * interface as constants.
```

Comment:
```suggestion
	 * class as constants.
```

`interface` or `class` ? (see my other remark about this elsewhere)